lib/dev.ts

```diff
@@ -236,3 +253,21 @@ async function generateDevJson(src: string, config: BaseConfig): Promise<DevJson
 
   return devJson;
 }
+
+export function buildGatewayObject(commandGateway, configGateway) {
+	// Gateway logic:
+	// if --no-gateway is provided (or commandGateway = false), gateway should be disabled ("");
+	// if -g is true, gateway is enabled, takes bundleUrl and tagName from `bos.config.json`
+	// if noone of the above option are specified, gateway should be the default gateway object;
+	const gatewayObject = DEFAULT_GATEWAY;
+
+	gatewayObject.enabled = commandGateway;
+
+  if (configGateway?.bundleUrl)
```

Comment:
Considering we have a default config, ability to override this config, or --no-gateway to disable it, I'm not entirely sure what this function does anymore. Take a step back and reassess it's purpose and if it's clearer to just check if gateway is disabled or not

Although we do need error checking -- we can't have bundleUrl without tagName and can't have tagName without bundleUrl

Reply:
(and let's have a test to confirm both must be populated)

Reply:
The error checking is in `config.ts` in the configuration schema, if I try to run the app without a gateway with both `tagName` and `bundleUrl` it will return this error: 

```
(node:20505) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
✗ SyntaxError: bos.config.json: Expected double-quoted property name in JSON at position 421 (line 14 column 2)
    at JSON.parse (<anonymous>)
    at _readFile (/Users/stefano/Workspace/Work/NEAR/bos-workspace/node_modules/jsonfile/index.js:25:16)
```

Without the option `-g` the logic is simpler but still needs to be expressed somewhere, that's the purpose of that function, somehow we need to specify the logic of the gateway: if it's disabled, if it's enabled but without a configuration object in `bos.config.json` or if it's enabled with a correctly formatted configuration object!

Let me know if you want to introduce more error checking on top of the Joi schema or if I need to move the gateway logic somewhere else!